File: ASP.NET Core + Angular/Task/OnlineBookStoreAPI/Models/Domain/CartItem.cs
namespace OnlineBookStoreAPI.Models.Domain
{
    public class CartItem
    {
        public int Id { get; set; }
        public bool Checked { get; set; }
        public int Quantity { get; set; }
        //public double TotalPrice { get; set; }
        public Cart Cart { get; set; }
        public Book Book { get; set; }
    }
}

File: ASP.NET Core + Angular/Task/client/src/app/_models/cartItem.ts
import { Book } from "./book";

export class CartItem {
  public book: Book;
  public qty: number;
  public checked: boolean;
  public totalPrice: number;
  
  constructor(book: Book, qty: number, totalPrice: number, checked: boolean) {
    this.book = book;
    this.qty = qty;
    this.checked = checked;
    this.totalPrice = totalPrice;
  }
}